Flaky-test: NarUnpackerTest.shouldExtractFilesOnceInDifferentProcess

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failures

- [2023-09-29T21:07:56.3124841Z](https://github.com/apache/pulsar/actions/runs/6354862390/job/17267802873#step:10:1657) 
- [2023-09-29T19:20:48.1844983Z](https://github.com/apache/pulsar/actions/runs/6354862390/job/17265259914#step:10:1702) 


### Exception stacktrace

```
java.lang.AssertionError: expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:110)
	at org.testng.Assert.failNotEquals(Assert.java:1577)
	at org.testng.Assert.assertTrue(Assert.java:56)
	at org.testng.Assert.assertTrue(Assert.java:66)
	at org.apache.pulsar.common.nar.NarUnpackerTest.shouldExtractFilesOnceInDifferentProcess(NarUnpackerTest.java:150)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

```


### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-common/src/test/java/org/apache/pulsar/common/nar/NarUnpackerTest.java
+++ b/pulsar-common/src/test/java/org/apache/pulsar/common/nar/NarUnpackerTest.java
@@ -38,7 +38,7 @@ import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 @Slf4j
-@Test(groups = "flaky")
+@Test
 public class NarUnpackerTest {
     File sampleZipFile;
     File extractDirectory;
@@ -47,7 +47,7 @@ public class NarUnpackerTest {
     public void createSampleZipFile() throws IOException {
         sampleZipFile = Files.createTempFile("sample", ".zip").toFile();
         try (ZipOutputStream out = new ZipOutputStream(new FileOutputStream(sampleZipFile))) {
-            for (int i = 0; i < 10000; i++) {
+            for (int i = 0; i < 5000; i++) {
                 ZipEntry e = new ZipEntry("hello" + i + ".txt");
                 out.putNextEntry(e);
                 byte[] msg = "hello world!".getBytes(StandardCharsets.UTF_8);
@@ -59,12 +59,20 @@ public class NarUnpackerTest {
     }
 
     @AfterMethod(alwaysRun = true)
-    void deleteSampleZipFile() throws IOException {
-        if (sampleZipFile != null) {
-            sampleZipFile.delete();
+    void deleteSampleZipFile() {
+        if (sampleZipFile != null && sampleZipFile.exists()) {
+            try {
+                sampleZipFile.delete();
+            } catch (Exception e) {
+                log.warn("Failed to delete file {}", sampleZipFile, e);
+            }
         }
-        if (extractDirectory != null) {
-            FileUtils.deleteFile(extractDirectory, true);
+        if (extractDirectory != null && extractDirectory.exists()) {
+            try {
+                FileUtils.deleteFile(extractDirectory, true);
+            } catch (IOException e) {
+                log.warn("Failed to delete directory {}", extractDirectory, e);
+            }
         }
     }
 
@@ -112,7 +120,7 @@ public class NarUnpackerTest {
 
     @Test
     void shouldExtractFilesOnceInDifferentProcess() throws InterruptedException {
-        int processes = 10;
+        int processes = 5;
         String javaExePath = findJavaExe().getAbsolutePath();
         CountDownLatch countDownLatch = new CountDownLatch(processes);
         AtomicInteger exceptionCounter = new AtomicInteger();
@@ -123,7 +131,9 @@ public class NarUnpackerTest {
                     // fork a new process with the same classpath
                     Process process = new ProcessBuilder()
                             .command(javaExePath,
-                                    "-Xmx64m",
+                                    "-Xmx96m",
+                                    "-XX:TieredStopAtLevel=1",
+                                    "-Dlog4j2.disable.jmx=true",
                                     "-cp",
                                     System.getProperty("java.class.path"),
                                     // use NarUnpackerWorker as the main class
@@ -131,6 +141,7 @@ public class NarUnpackerTest {
                                     // pass arguments to use for testing
                                     sampleZipFile.getAbsolutePath(),
                                     extractDirectory.getAbsolutePath())
+                            .redirectErrorStream(true)
                             .start();
                     String output = IOUtils.toString(process.getInputStream(), StandardCharsets.UTF_8);
                     int retval = process.waitFor();
@@ -148,7 +159,7 @@ public class NarUnpackerTest {
                 }
             }).start();
         }
-        assertTrue(countDownLatch.await(30, TimeUnit.SECONDS));
+        assertTrue(countDownLatch.await(30, TimeUnit.SECONDS), "All processes should finish before timeout");
         assertEquals(exceptionCounter.get(), 0);
         assertEquals(extractCounter.get(), 1);
     }

